src/app/shared/operators/state.spec.ts

```diff
@@ -130,9 +137,65 @@ describe('mapToState', () => {
   it('should retry even after a source failure', () => {
     testScheduler.run(helpers => {
       const { cold, expectObservable } = helpers;
-      const e1 =  cold('-e|');
-      const res = cold('----b--b--|');
-      const expected = 'fe--fe-fe-|';
+      const e1 =  cold('-#|');
+      const res = cold('----0--0--|');
+      const expected = 'fx--fx-fx-|';
+
+      expectObservable(e1.pipe(
+        mapToFetchState({ resetter: res }),
+        stateToLetter(),
+      )).toBe(expected);
+    });
+  });
+
+  it('should allow several fetching states with no data', () => {
+    testScheduler.run(helpers => {
+      const { cold, expectObservable } = helpers;
+      const e1 =  cold('---1|');
+      const res = cold('-0------|'); // the values are not important for the resetter
+      const expected = 'ff--a---|';
+
+      expectObservable(e1.pipe(
+        mapToFetchState({ resetter: res }),
+        stateToLetter(),
+      )).toBe(expected);
+    });
+  });
+
+  it('should pass data between fetching states', () => {
+    testScheduler.run(helpers => {
+      const { cold, expectObservable } = helpers;
+      const e1 =  cold('--1|');
+      const res = cold('----00----|'); // the values are not important for the resetter
+      const expected = 'f-a-gg-a--|';
+
+      expectObservable(e1.pipe(
+        mapToFetchState({ resetter: res }),
+        stateToLetter(),
+      )).toBe(expected);
+    });
+  });
+
+  it('should not replay the last data after a failure', () => {
+    testScheduler.run(helpers => {
+      const { cold, expectObservable } = helpers;
+      const e1 =  cold('-1-#');
+      const res = cold('-------0------|');
+      const expected = 'fa-x---fa-x---|';
+
+      expectObservable(e1.pipe(
+        mapToFetchState({ resetter: res }),
+        stateToLetter(),
+      )).toBe(expected);
+    });
+  });
+
+  it('should change the data in state as expected', () => {
+    testScheduler.run(helpers => {
+      const { cold, expectObservable } = helpers;
+      const e1 =  cold('--1--2|');
+      const res = cold('----0------0-----|'); // the values are not important for the resetter
+      const expected = 'f-a-g-a--b-h-a--b|';
```

Comment:
Although the mapping letter/state is quite readable, as the test is growing the same mapping becomes harder and harder to read, and probably to maintain too.
I think it would be worth looking for another way, digging around the second arguments of `cold()` and `expectObservable(…).toBe()`
As an example, it could give something like this:
```ts
const e1 =  cold('--1--2', { 1: 'hello', 2: 'world' });
const res = cold('--0---0-'); // values are not important
const expected = 'f-a---g-a--';
expectObservable(e1.pipe(
  mapToFetchState(),
  map(state => `tag: ${state.tag}, value: ${String(state.data)}`), // any serialization does the trick.
)).toBe(expected, {
  f: 'tag: fetching, value: undefined',
  g: 'tag: fetching, value: hello',
  a: 'tag: ready, value: hello world',
});
```
(Note: The test is invalid, it's a quick sample.)

That way, there would be no need to know the whole letter/state mapping, each test would encapsulate its input/output expectation.

What do you think of the general idea ?